pkg/kubelet/kubelet.go

```diff
@@ -1899,6 +1901,9 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 			handler.HandlePodSyncs([]*v1.Pod{pod})
 		}
 	case <-housekeepingCh:
+		if len(os.Getenv("WATCHDOG_USEC")) > 0 {
+			daemon.SdNotify(false, daemon.SdNotifyWatchdog)
+		}
```

Comment:
I'm puzzled; I'd expect to see the value from `os.Getenv("WATCHDOG_USEC")` used to calculate a period for a timer - usually you divide by 2.

Typical implementations run some code before the watchdog expires to check that the main event loop is not stuck, and then (satisfied that the app is not obviously broken) notify the supervisor to reset its watchdog timer.

Reply:
I worry that this could result in a lot more notifies than systemd was expecting / designed for.

Reply:
> I worry that this could result in a lot more notifies than systemd was expecting / designed for.

Given that the threshold is in microseconds, I don't anticipate any issues notifying systemd every 2s.

> I'm puzzled; I'd expect to see the value from os.Getenv("WATCHDOG_USEC") used to calculate a period for a timer - usually you divide by 2.

I was going for minimally invasive, and `housekeepingPeriod` seemed sufficiently short, but I can create a dedicated channel derived from `WATCHDOG_USEC` instead (although at least x/3 would be preferable - we want to be *really* sure there is actually a problem at the point we initiate recovery).

> Typical implementations run some code before the watchdog expires to check that the main event loop is not stuck

Perhaps I am fundamentally misunderstanding golang or `syncLoopIteration()`, but IIUC `SdNotify()` would not be called if the main event loop was stuck.  Making the proposed code a sufficient indicator that kubelet is not hung. 

Deciding whether to expand "health" to non-hang scenarios (such as if `kl.sourcesReady.AllReady()` remains untrue for "too long") is beyond the scope of what I wish to achieve at this time.

Reply:
Ok, so the main challenge is if the sysadmin configures the watchdog with an interval of say 1s and that's too short for the housekeeping interval to keep the watchdog alive.
(and for me it's also more about the behavior being confusing and hard to document, than that a 6s watchdog interval is unacceptable once you know that's a plausible minimum).